news/utils.py

```diff
@@ -0,0 +1,138 @@
+import re
+import requests
+from bs4 import BeautifulSoup, NavigableString
+from usp.tree import sitemap_tree_for_homepage
+
+def build_url_index(base_url, keyword=None):
+
+    if keyword is None:
+        keyword = [""]
+    elif isinstance(keyword, str):
+        keyword = [keyword]
+
+    tree = sitemap_tree_for_homepage(base_url)
+    page_index = [page.url for key in keyword for idx, page in enumerate(tree.all_pages()) if key in page.url]
```

Comment:
What is the `idx` and enumeration for here? It doesn't seem to be used/saved anywhere